examples/client/client/tasks.nim

```diff
@@ -610,6 +622,8 @@ proc login*(account: int, password: string) {.
 
     statusState = StatusState.loggedin
 
+    asyncSpawn updatePrices()
```

Comment:
Thanks for a great observation! 

What if we simplify the `updatePrices` proc so that it looks like this:
```
proc updatePrices() {.async.} =
  while statusState == StatusState.loggedin:
    try:
      await status.updatePrices()
    except CatchableError as e:
      trace "updatePrices error:", err=e.msg
    await sleepAsync(updatePricesTimeout)
```
So it will run in the background, periodically trying to re-fetch prices if user is logged in currently. I think it's fine that the updater is started during the first login, and i've moved the `sleepAsync` to the end of the loop body, as you suggested. The prices are not account-dependent anyway.

I think i've accidentally clicked `Resolve conversation` on one of your comments, but it pertained to `updatePrices` anyway i think.

Reply:
I reviewed all the changes made and they look good. Thanks for adding the `/setpricetimeout` command, very nice!

However, there's still the problem of multiple updater loops being spawned.

Take a look at these changes: https://github.com/status-im/nim-status/compare/c26f0321b24aa5d5f4f010851054b4a7984c616c...feat/token-prices-2.

In that branch I've rebased on top of my `fix/help` branch for PR #277, made same changes for commands you introduced, and then I added some changes to `examples/client/client/tasks.nim` to demonstrate the problem and proposed fix.

The code for the proposed fix is commented out.

Note that I also removed the `loggedin` check in `proc setPriceTimeout` in `client/tasks.nim` since there's no good reason to have it.

To see the problem manifest:
* Build and run the client, then `/setpricetimeout 20`, `/login ...`, wait until login is successful, and then `/logout` and `/login ...` several times before 20 seconds is up.
* Watch the log: in intervals of 20 seconds you'll see that multiple loop "instances" are running.

If you uncomment the proposed fix (lines involving threadvar `updatingPrices`) and repeat the experiment, there will only be one instance of the loop running at any given time when `loggedin`.

---

Unrelated to the above, I see that an exception is being raised every time the loop runs; in the log it looks like:

```
TRC 2021-08-17 13:56:14.750-05:00 Exception when parsing price API response  topics="client" tid=16054077 file=tokens.nim:139 err="key not found: RAW"
```

Reply:
Regarding the exception - it's a normal flow in case when the URL is malformed (e.g. when we have no tokens to fetch prices for). Maybe this can be handled by a conditional in another part of code.